Add Badge Specification (#131)
pulsar-edit/package-backend: docs/reference/badge-spec.md

# Pulsar Package's Badges

> A badge allows extra information to be communicated about that package. Similar to how NPM may warn of errors after installing a package, or how a user is able to receive badges on their GitHub profile.

Reviewer: ```suggestion
A badge allows extra information to be communicated to a Pulsar user about a package. Similar to how NPM may warn of errors after installing a package, or how a user is able to receive badges on their GitHub profile.
```
Just fleshed out a little to indicate the audience of the badges.

A badge allows the Pulsar admins to communicate certain information about a package quickly to end users everywhere it may appear.

Below is the specification for a badge, the data it may contain, the valid values that may be placed in an enum, and what should be done with it.

## Schema

The following schema is required of any `generic` badge type:

```json
"badges": [
  {
    "type": "<type_enum>",
    "title": "<string>",
    "text": "<string>",
    "link": "<string>"
  }
]
```

> As you can see `badges` is an array of a valid `badge` object.

Reviewer: ```suggestion
As you can see `badges` is an array of a valid `badge` objects.
```
?

Any `badge` object may contain the following properties:

* `title`: This is a **required** property. Specifying the title of the badge. Such as `Deprecated`.
* `text`: This is an **optional** property. Specifying further information about this badge. That may be hidden either based on a user setting or by context.
* `link`: This is an **optional** property. Providing a link for the user to be directed to, that may contain further information about the badge.

> * `type`: This is a **required** property. Instructing what type of badge this is. The type of badge should be used to determine any icons used alongside the badge, as well as any colour or styling the badge will receive when displayed by the client. The valid `type`s currently available are as follows:

Reviewer: ```suggestion
* `type`: This is a **required** property. Instructing what type of badge this is. The type of badge should be used to determine any icons used alongside the badge, as well as any color or styling the badge will receive when displayed by the client. The valid `type`s currently available are as follows:
```

As much as it pains me to do it, this should be in AmE.

>   - `warn`: This should be used to indicate an error that the user **must** be aware of.

Reviewer: ```suggestion
  - `warn`: This should be used to indicate something that the user **must** be aware of.
```
It isn't necessarily an error as such as it could convey different kinds of info that is essential to know but not necessarily breaking. Not sure about my wording, just `error` doesn't quite feel correct.
